src/main/java/com/google/sps/servlets/NewTripServlet.java

```diff
@@ -0,0 +1,73 @@
+
+package com.google.sps.servlets;
+
+import com.google.protobuf.Timestamp;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.gson.Gson;
+import com.google.sps.model.Trip;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+ 
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.text.StringEscapeUtils;
+
+@WebServlet("/NewTrip")
+public class NewTripServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String error = validateInput(request);
+    if (error.isEmpty()) {
+    // Get the value entered in the form.
+    String textValuetitle = StringEscapeUtils.escapeHtml4(request.getParameter("text-input-title"));
+    float totalBudget = Float.parseFloat(StringEscapeUtils.escapeHtml4(request.getParameter("text-input-totalBudget")));
+    
```

Comment:
No need to escape the float. ParseFloat will throw and exception, which ideally the client should catch and do something.